hnn_core/network.py

```diff
@@ -466,6 +466,27 @@ def update_types(self, gid_dict):
             spike_types += [list(spike_types_trial)]
         self._types = spike_types
 
+    def get_spike_rates(self):
+        """Spike rates by cell type.
+
+        Returns
+        -------
+        spike_rate : dict
+            Dictionary with keys 'L5_pyramidal', 'L5_basket', etc.
+        """
+        cell_types = ['L5_pyramidal', 'L5_basket', 'L2_pyramidal', 'L2_basket']
+        spike_rates = dict()
+        all_spike_times = np.array(sum(self._times, []))
+        tstart, tstop = min(all_spike_times), max(all_spike_times)
```

Comment:
one complication that creeps up here is that we don't have access to the start and end time of the trials in the spike object. Perhaps that needs to be an attribute of the spikes object. Or perhaps we need another "times" array which stores the time instances of simulation and a different "spike_times" arrange which tells us when in the "times" array we get a spike.